src/sources/postgresql_metrics.rs

```diff
@@ -0,0 +1,851 @@
+use crate::{
+    config::{DataType, GlobalOptions, SourceConfig, SourceDescription},
+    event::metric::{Metric, MetricKind, MetricValue},
+    internal_events::{PostgresqlMetricsCollectCompleted, PostgresqlMetricsCollectFailed},
+    shutdown::ShutdownSignal,
+    Event, Pipeline,
+};
+use chrono::{DateTime, Utc};
+use futures::{
+    future::{join_all, try_join_all},
+    stream, FutureExt, SinkExt, StreamExt, TryFutureExt,
+};
+use openssl::{
+    error::ErrorStack,
+    ssl::{SslConnector, SslMethod},
+};
+use postgres_openssl::MakeTlsConnector;
+use serde::{Deserialize, Serialize};
+use snafu::{ResultExt, Snafu};
+use std::{
+    collections::{BTreeMap, HashSet},
+    future::ready,
+    path::PathBuf,
+    time::Instant,
+};
+use tokio::time;
+use tokio_postgres::{
+    config::{ChannelBinding, Host, SslMode, TargetSessionAttrs},
+    types::FromSql,
+    Client, Config, Error as PgError, NoTls, Row,
+};
+
+macro_rules! tags {
```

Comment:
Another one! :smile: Linking to https://github.com/timberio/vector/issues/4651 to ensure we swap this one too.